scripts/cleanup-bcd-overrides.ts

```diff
@@ -0,0 +1,114 @@
+import { setLogger } from "compute-baseline";
+import { Compat, Feature } from "compute-baseline/browser-compat-data";
+import { fdir } from "fdir";
+import fs from "node:fs";
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+import { isDeepStrictEqual } from "node:util";
+import winston from "winston";
+import YAML from "yaml";
+import yargs from "yargs";
+import { checkForStaleCompat } from "./dist";
+
+const compat = new Compat();
+
+const argv = yargs(process.argv.slice(2))
+  .scriptName("cleanup-bcd-overrides")
+  .usage(
+    "$0 [paths..]",
+    "Remove `compat_features` from `.yml` files that have an equivalently tagged set of features in @mdn/browser-compat-data",
+    (yargs) =>
+      yargs.positional("paths", {
+        describe: "Directories or files to check/update.",
+        default: ["features"],
+      }),
+  ).argv;
+
+const logger = winston.createLogger({
+  level: argv.verbose > 0 ? "debug" : "warn",
+  format: winston.format.combine(
+    winston.format.colorize(),
+    winston.format.simple(),
+  ),
+  transports: new winston.transports.Console(),
+});
+
+let exitStatus = 0;
+
+setLogger(logger);
+
+const tagsToFeatures: Map<string, Feature[]> = (() => {
+  // TODO: Use Map.groupBy() instead, when it's available
+  const map = new Map();
+  for (const feature of compat.walk()) {
+    for (const tag of feature.tags) {
+      let features = map.get(tag);
+      if (!features) {
+        features = [];
+        map.set(tag, features);
+      }
+      features.push(feature);
+    }
+  }
+  return map;
+})();
+
+function cleanup(sourcePath: string): void {
+  const source = YAML.parseDocument(
+    fs.readFileSync(sourcePath, { encoding: "utf-8" }),
+  );
+  const { name: id } = path.parse(sourcePath);
+
+  // Collect tagged compat features. A `compat_features` list in the source
+  // takes precedence, but can be removed if it matches the tagged features.
+  const taggedCompatFeatures = (tagsToFeatures.get(`web-features:${id}`) ?? [])
+    .map((f) => `${f.id}`)
+    .sort();
+
+  const data = (source.contents as any).get("compat_features");
```

Comment:
There's one more place that the comment is removed, before the `compat_features` key. A good example is the `article.yml` feature. I think we can get that key by changing this to-
```suggestion
    const { key: keyData, value: data } = (source.contents as any).items.find(
    (p) => p.key.value === "compat_features",
  );
```

And then down below, it'll need to be something like-

```
      const comments = keyData.commentBefore ? [keyData.commentBefore] : [];
      if (data.commentBefore) comments.push(data.commentBefore);
```

Although it does appear that this adds some unneeded whitespace.